Makefile

```diff
@@ -144,6 +144,42 @@ ifeq ($(UNAME_M),$(filter $(UNAME_M),x86_64 i686 amd64))
 			CXXFLAGS += -mavx2
 		endif
 
+		AVX512F_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512F')
+		ifneq (,$(AVX512F_M))
+			CFLAGS   += -mavx512f
+			CXXFLAGS += -mavx512f
+		endif
+
+		AVX512DQ_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512DQ')
+		ifneq (,$(AVX512DQ_M))
+			CFLAGS   += -mavx512dq
+			CXXFLAGS += -mavx512dq
+		endif
+
+		AVX512CD_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512CD')
+		ifneq (,$(AVX512CD_M))
+			CFLAGS   += -mavx512cd
+			CXXFLAGS += -mavx512cd
+		endif
+
+		AVX512CD_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512BW')
+		ifneq (,$(AVX512BW_M))
+			CFLAGS   += -mavx512bw
+			CXXFLAGS += -mavx512bw
+		endif
+
+		AVX512VL_M := $(shell $(CPUINFO_CMD) | grep -iw 'AVX512VL')
+		ifneq (,$(AVX512VL_M))
+			CFLAGS   += -mavx512vl
+			CXXFLAGS += -mavx512vl
+		endif
+
+		AVX512VNNI_M := $(shell $(CPUINFO_CMD) | grep -iwE 'AVX512_VNNI|AVX512VNNI')
```

Comment:
AVX512_VNNI is for Linux, so just out of curiosity:
Which OS uses AVX512VNNI?

As you're adding VNNI beside basic AVX-512 subsets, I would suggest adding also VBMI, as llama.cpp does it too. It's nice to have some kind of parity between these projects.

Moreover, if you're adding VNNI and VBMI in Makefile, you should add them in CMakeLists.txt too, via `WHISPER_NO_AVX512_VNNI` and `WHISPER_NO_AVX512_VBMI`, defaulting to `ON`. I suggest using `add_compile_definitions` similarly like llama.cpp does.

Reply:
Added AVX512VNNI for pure symmetry just in case any system does have it.

Reply:
I asked about AVX512VNNI, didn't request removing. But I see you removed it in dc69460b42.

So I looked around and it seems AVX512VNNI is the name reported by FreeBSD.
http://fxr.watson.org/fxr/source/x86/x86/identcpu.c#L994

Please bring back `grep -iwE 'AVX512_VNNI|AVX512VNNI'` for VNNI detection.

Reply:
Great, my feeling was right then.